assets/prototype/application/ui/components/display/entityListFilterBar/useEntityListFilterState.js

```diff
@@ -0,0 +1,92 @@
+import { useReducer, useEffect } from '@wordpress/element';
+
+const useEntityListFilterState = () => {
+	const initialState = {
+		perPage: 6,
+		searchText: '',
+		showEntityFilters: false,
+		view: 'list',
+	};
+	const [state, dispatch] = useReducer(reducer, initialState);
+
+	useEffect(() => {
+		console.log('useEntityListFilterState >>>', state);
+	}, [state]);
+
+	const setGridView = () => {
+		dispatch({
+			type: 'SET_VIEW',
+			view: 'grid',
+		});
+	};
+
+	const setListView = () => {
+		dispatch({
+			type: 'SET_VIEW',
+			view: 'list',
+		});
+	};
+
+	const setPerPage = (perPage) => {
+		dispatch({
+			perPage,
+			type: 'SET_PER_PAGE',
+		});
+	};
+
+	const setSearchText = (searchText) => {
+		dispatch({
+			searchText,
+			type: 'SET_SEARCH_TEXT',
+		});
+	};
+
+	const toggleEntityFilters = () => {
+		const showEntityFilters = !state.showEntityFilters;
```

Comment:
This operation should goto the reducer IMHO.

Reply:
done, thx